ibis/impala/compiler.py

```diff
@@ -138,6 +141,16 @@ def _cast(translator, expr):
 def _type_to_sql_string(tval):
     if isinstance(tval, dt.Decimal):
         return 'decimal({}, {})'.format(tval.precision, tval.scale)
+    if isinstance(tval, dt.Map):
+        return 'map<{}, {}>'.format(
+            _type_to_sql_string(tval.key_type),
+            _type_to_sql_string(tval.value_type)
+            )
+    if isinstance(tval, dt.Array):
+        return 'array<{}>'.format(_type_to_sql_string(tval.value_type))
+    if isinstance(tval, dt.Struct):
+        parts = ["{}:{}".format(n, _type_to_sql_string(t)) for n, t in tval.pairs.items()] # noqa E501
```

Comment:
Can you format the list comprehension instead of ignoring the warning?

Reply:
I did look at reformatting but the list comprehension across the multiple lines gets harder to read so i went for clarity.  Happy to change if this is the preference.

Reply:
Actually, you can leave it as is except remove the `noqa` comment since the line fits within the `flake8` margin.